docs/faqs.md

```diff
@@ -0,0 +1,153 @@
+
+## Why not format X?
+
+### Why not OpenQASM2?
+
+[OpenQASM2](https://arxiv.org/abs/1707.03429) has become the de-facto interchange format in quantum software, but the biggest drawback
+has been the lack of program structure — something that we *need* to be a first-class citizen in a
+compiler exchange format.
+
+An exchange format that makes it easy to incorporate structure will also allow the field to consider
+compilation of structured programs by default.
+
+### Why not OpenQASM3?
+
+[OpenQASM3](https://arxiv.org/abs/2104.14722) brings program structure to OpenQASM2, however has not yet seen adoption or tooling
+maturity in the ecosystem to the extent as OpenQASM2. In addition, OpenQASM3 maintains a focus on
+human readability (requiring string generation and parsing), and is not designed first and foremost for quantum optimization.
+
+Moreover, OpenQASM3 has a few quirks that limit its general usability as a compiler exchange
+format:
+
+- Lack of extensibility without evolving the specification (a `#pragma` directive is provided, but
+  does not easily allow for including general structured information).
+
+- You can only index into arrays using compile-time constants, not dynamic values that may only been
+  known at runtime.
+
+- Registers (TODO).
+
+### Why not QIR?
+
+[QIR](https://www.qir-alliance.org/) was originally designed as a low-level, hardware-facing IR for optimization, for example using
+the QIR Adapter Tool (QAT). However, QIR today is used less for quantum optimization as it is quite
+low-level (missing a lot of the higher-level abstractions we would like to preserve during
+compilation), and is predominantly used as a low-level interchange format between software
+frameworks and hardware providers.
+
+This can be seen by looking at the various software compilers built on top on the QIR stack such as
+[CUDA-Q](https://github.com/nvidia/cuda-quantum) and [Catalyst](https://github.com/pennylaneai/catalyst) — rather than compiling QIR programs, they instead
+optimize MLIR, a higher-level intermediate representation.
+
+Furthermore, relying on QIR as a compilation exchange format introduces a big barrier to entry due
+to it depending on the LLVM toolchain — a massive software project that has non-trivial
+compatibility concerns, installation, and learning overhead.
+
+### Why not MLIR?
+
+[MLIR (Multi-Level Intermediate Representation)](https://en.wikipedia.org/wiki/MLIR_(software)) is a framework that exists on top of QIR/LLVM, and
+permits multiple high-level 'dialects' (IRs tailored to specific domains) to coexist, making it the
+framework of choice for most QIR/LLVM compiler stacks such as CUDA-Q and Catalyst.
+
+However, something to note is that MLIR is a *framework*, not a defined format — different compilers
+define *multiple* internal/custom dialects that altogether are used to represent a quantum program.
+At the moment, there is no clear *shared* set of quantum MLIR dialects that make sense to use more
```

Comment:
An additional point to mention is the difficulty in maintaining binary compatibility between different projects, even if they did share the same mlir dialect. The textual mlir format would avoid this though.

Reply:
https://github.com/unitaryfoundation/jeff/pull/10/files#r2312563136